src/cocotb/task.py

```diff
@@ -152,6 +151,9 @@ def _advance(self, outcome: Outcome) -> Any:
         except StopIteration as e:
             self._outcome = Value(e.value)
             self._state = Task._State.FINISHED
+        except CancelledError as e:
+            self._outcome = Error(remove_traceback_frames(e, ["_advance", "send"]))
+            self._state = Task._State.CANCELLED
```

Comment:
Doesn't this allow the task to suppress the `CancelledError`, if it's only set as `CANCELLED` after the task raises a `CancelledError`?
We need the `CancellationError` from #3874